prover/crypto/vortex/commitment.go

```diff
@@ -31,7 +28,9 @@ type EncodedMatrix []smartvectors.SmartVector
 //
 // And can be safely converted to a field Element via
 // [field.Element.SetBytesCanonical]
-func (p *Params) CommitMerkle(ps []smartvectors.SmartVector) (encodedMatrix EncodedMatrix, tree *smt.Tree, colHashes []field.Element) {
+// We apply SIS+MiMC hashing on the columns to compute leaves only if
+// isSISAppliedForRound is true, otherwise we apply only the MiMC hashing.
+func (p *Params) CommitMerkle(ps []smartvectors.SmartVector, isSISAppliedForRound bool) (encodedMatrix EncodedMatrix, tree *smt.Tree, colHashes []field.Element) {
```

Comment:
Better to make two separate function and factor out the common parts, no?

Reply:
Ah, I see. You basically reused the code for replaceBySis.

Reply:
But at this point, it is cleaner to have two function as this is a public function